src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -127,5 +128,19 @@ internal static MultiClusterRegistrationStrategy GetMultiClusterRegistrationStra
                             grainClass.Name));
             }
         }
+
+        // used by Orleankka's bootstrapper
+        internal void SetMessageAlwaysInterleaves(Func<object, bool> predicate)
```

Comment:
it would be good to combine this with @sergeybykov's suggestion of being able to define it statically, with an attribute or something on the grain implementation or interface, or even on the argument type as defined in the interfaces that the silo uses. I would prefer on the grain type itself, since the argument type could be re-used across grain implementations, and each have different interleaving implications, or even different grain implementations for the same grain interface might have different needs.